Userland/Services/FileSystemAccessServer/ClientConnection.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2021, timmot <tiwwot@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "DownloadWindow.h"
+#include <AK/Debug.h>
+#include <FileSystemAccessServer/ClientConnection.h>
+#include <LibC/fcntl.h>
+#include <LibCore/File.h>
+#include <LibCore/StandardPaths.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/FilePicker.h>
+#include <LibGUI/MessageBox.h>
+
+namespace FileSystemAccessServer {
+
+static HashMap<int, RefPtr<ClientConnection>> s_connections;
```

Comment:
```suggestion
static HashMap<int, NonnullRefPtr<ClientConnection>> s_connections;
```
As a rule of thumb, you (almost) never want to store null pointers in a `HashMap`, `Vector` or similar.